### PR TITLE
Teal refactor

### PR DESCRIPTION
**Pull request by nikolas-burkoff**
Part of https://github.com/insightsengineering/teal/issues/731 - will need core team input as well as Nick P's

### CODE REVIEW

File: R/utils.R [hunk 1]
```diff
@@ -224,7 +219,8 @@ constr_anl_chunks <- function(session, input, datasets, dataname, param_id, para
     validate(need(constraint_var, "select a constraint variable"))
 
     # note that filtered is false thus we cannot use anl_param()$ANL
```
Comment (nikolas-burkoff) [inline R/utils.R:221]:
```suggestion
    
```

File: R/utils.R [hunk 2]
```diff
@@ -223,8 +218,9 @@ constr_anl_chunks <- function(session, input, datasets, dataname, param_id, para
     constraint_var <- input[["constraint_var"]]
     validate(need(constraint_var, "select a constraint variable"))
 
-    # note that filtered is false thus we cannot use anl_param()$ANL
-    ANL <- datasets$get_data(dataname, filtered = FALSE) # nolint
+
```
Comment (nikolas-burkoff) [inline R/utils.R:221]: @npaszty we can no longer use non-filtered data in the modules I don't think this has an impact but please get in touch ASAP if it does

File: R/tm_g_gh_lineplot.R [hunk 3]
```diff
@@ -789,22 +805,24 @@ srv_lineplot <- function(id,
...
         card
       }
       teal.reporter::simple_reporter_srv("simple_reporter", reporter = reporter, card_fun = card_fun)
     }
     ###
 
-    get_rcode_srv(
+    teal.widgets::verbatim_popup_srv(
+      id = "warning",
+      verbatim_content = reactive(teal.code::get_warnings(plot_q)),
```
Comment (nikolas-burkoff) [inline R/tm_g_gh_lineplot.R:817]:
```suggestion
      verbatim_content = reactive(teal.code::get_warnings(plot_q())),
```

**Review by nikolas-burkoff — COMMENTED** [review]
I can't approve my own PR but I'm happy with this now 🏆

File: R/utils.R [hunk 4]
```diff
@@ -282,7 +277,7 @@ constr_anl_chunks <- function(session, input, datasets, dataname, param_id, para
 
       update_min_max(session, args)
     } else {
-      stop("invalid contraint_var", constraint_var)
+      validate(need(FALSE, "This is an invalid data contraint for the filtered data"))
```
Comment (nikolas-burkoff) [inline R/utils.R:280]: As we are using filtered not unfiltered data here this needs to be a validate not a stop

File: R/tm_g_gh_boxplot.R [hunk 5]
```diff
@@ -384,14 +389,14 @@ srv_g_boxplot <- function(id,
...
+      reactive_ANL = anl_q
     )
-    keep_data_const_opts_updated(session, input, anl_chunks, "xaxis_param")
+    keep_data_const_opts_updated(session, input, anl_q, "xaxis_param")
 
     horizontal_line <- srv_arbitrary_lines("hline_arb")
 
     create_plot <- reactive({
-      private_chunks <- teal.code::chunks_deep_clone(anl_chunks()$chunks)
+      req(anl_q())
```
Comment (gogonzo) [inline R/tm_g_gh_boxplot.R:399]: good bye` chunks_deep_clone` ❤️

**Review by gogonzo — APPROVED** [review]
Followup issues after spending couple of hours in the encodings code

https://github.com/insightsengineering/teal.goshawk/issues/204
https://github.com/insightsengineering/teal.goshawk/issues/205

**Pull request merged**